### PR TITLE
[ROCm] ConvBfloat16Support HLO pass for AMDGPU Compiler

### PR DESCRIPTION
**Pull request by hsharsha**

### CODE REVIEW

File: xla/service/gpu/amdgpu_compiler.cc [hunk 1]
```diff
@@ -63,6 +94,12 @@ absl::Status AMDGPUCompiler::OptimizeHloConvolutionCanonicalization(
   pipeline.AddInvariantCheckerDebug<HloVerifier>(
       /*layout_sensitive=*/false,
       /*allow_mixed_precision=*/false);
+
+  // Convert upsupported bf16 convolutions to f32.
```
[the remark this answers is not in the document]
Comment (hsharsha) [reply xla/service/gpu/amdgpu_compiler.cc]: fixed.

**Review by d0k — APPROVED** [review]

**Comment by kamaljeeti** [comment]
Hi @ddunl , there is a check that is failing can you look into this once? Thanks.

**Comment by ddunl** [comment]
Sorry for delay here, can you resolve conflicts?

**Review by ddunl — APPROVED** [review]
Thanks for the very quick rebase!!

**Pull request closed without merging**